app/publishing/delinkedCalculation/get-unpublished-delinked.js

```diff
@@ -0,0 +1,48 @@
+const db = require('../../data')
+const { publishingConfig } = require('../../config')
+
+const getUnpublishedDelinked = async (transaction) => {
+  return db.delinkedCalculation.findAll({
+    lock: true,
+    skipLocked: true,
+    limit: publishingConfig.dataPublishingMaxBatchSizePerDataSource,
+    where: {
+      [db.Sequelize.Op.or]: [
+        {
+          datePublished: null
+        },
+        {
+          datePublished: { [db.Sequelize.Op.lt]: db.sequelize.col('updated') }
+        }
+      ]
+    },
+    attributes: [
+      ['applicationId', 'applicationReference'],
+      ['calculationId', 'calculationReference'],
```

Comment:
if we're stripping out all of the additional code Amirs had in place I don't know that we need this renaming of variable names any more. If you think about how this is working currently:

- Gets data with appId and calcId remapped to appRef and calcRef
- Identifies all fields with calcRef, and then maps the Ref fields back to Id
- Saves to d365 table, which has calcId and not calcRef

So... what do we actually need calcRef and appRef for? We don't use them as Ref before remapping to Id.

Reply:
Yes we need to alias calculationId into calculationReference, and applicationId to applicationReference, anything we use that has Id as the last two chars cannot pass through the system due to the restriction.